### PR TITLE
New Kotlin Cheat Sheet

### PR DESCRIPTION
**Pull request by mr-max**
IA: https://duck.co/ia/view/kotlin

### CODE REVIEW

**Comment by daxtheduck** [comment]
@mr-max Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please head to https://quackslack.herokuapp.com/ to get an invite.

>More Info: http://docs.duckduckhack.com/submitting/submitting-overview.html

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=kotlin cheat sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/kotlin_mobile.png)

File: share/goodie/cheat_sheets/json/kotlin.json [hunk 1]
```diff
@@ -0,0 +1,165 @@
+{
+  "id": "kotlin_cheat_sheet",
+  "name": "Kotlin",
+  "description": "Basic syntax, functions, and expressions of the Kotlin language",
+  "metadata": {
+    "sourceName": "Kotlin reference",
+    "sourceUrl": "https://kotlinlang.org/docs/reference/"
+  },
+  "aliases": [],
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/kotlin.json]: You can simply remove this if there are no aliases 👍

**Comment by moollaza** [comment]
@mr-max Thanks for submitting this! It LGTM :+1: and it's also live on Beta. Would you be interested in sharing this with the Kotlin community to get their feedback on it?

**Comment by daxtheduck** [comment]
@mr-max this is now merged! It should be live on DuckDuckGo within one week (or during the next release) and we'll be sure to notify you when it happens.

After your Instant Answer goes live for the World to use, you will be able to see [traffic information](http://docs.duckduckhack.com/maintaining/tools.html#monitor-your-instant-answer-page) about your IA as well as receive feedback directly from users. 
                
Please refer to the [maintenance guide and docs](http://docs.duckduckhack.com/maintaining/guidelines.html) to keep up with potential changes to your Instant Answer.
                
Thank you again for your help!

File: share/goodie/cheat_sheets/json/kotlin.json (hunk 1 above)
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/kotlin.json]: We should get rid of this seeing as it is empty.

File: share/goodie/cheat_sheets/json/kotlin.json [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+      {
+        "key": "var weather = \"sunny\"",
+        "val": "Define a non-null variable"
+      },
+      {
+        "key": "var weather : String = \"sunny\"",
+        "val": "Define a variable with an explicit non-null type"
+      },
+      {
+        "key": "var weather : String? = null",
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/kotlin.json]: We should be using 4 spaces per indent throughout.

File: share/goodie/cheat_sheets/json/kotlin.json [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+      },
+      {
+        "key": "a - b",
+        "val": "Create a collection of all items of a that do not occur in b"
+      }
+    ],
+    "Control structures": [
+      {
+        "key": "if (a < b) a else b",
+        "val": "Condition"
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/kotlin.json]: `Conditional`

File: share/goodie/cheat_sheets/json/kotlin.json [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+        "key": "class MyList : ArrayList() \\{ … \\}",
+        "val": "Inherit from another class or interface"
+      },
+      {
+        "key": "enum class Direction \\{ UP, DOWN \\}",
+        "val": "Define an enum"
+      },
+      {
+        "key": "data class Person(val name: String, val age: Int)",
+        "val": "Define a data class, with automatic equals(), hashCode() and toString()"
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/kotlin.json]: Use an Oxford comma.

**Pull request merged**